# Allow listen() to succeed if task has already been created
diff -Naur libraries/AsyncUDP/src/AsyncUDP.cpp /c/Users/trial/AppData/Local/Arduino15/packages/esp32/hardware/esp32/3.1.0/libraries/AsyncUDP/src/AsyncUDP.cpp
--- libraries/AsyncUDP/src/AsyncUDP.cpp	2024-12-29 00:22:07.829879300 +0000
+++ /c/Users/trial/AppData/Local/Arduino15/packages/esp32/hardware/esp32/3.1.0/libraries/AsyncUDP/src/AsyncUDP.cpp	2025-09-27 11:35:08.601713000 +0100
@@ -573,7 +573,7 @@
 }
 
 bool AsyncUDP::listen(const ip_addr_t *addr, uint16_t port) {
-  if (!_udp_task_start()) {
+  if (!_udp_task_handle && !_udp_task_start()) {
     log_e("failed to start task");
     return false;
   }
